php/Terminus/Models/Upstream.php

```diff
@@ -37,7 +45,7 @@ public function fetch(array $args = [])
     {
         $options = array_merge(['options' => ['method' => 'get',],], $this->args, $args);
         $results = $this->request->request($this->url, $options);
-        $this->attributes = $this->parseAttributes($results['data']->upstream);
+        $this->attributes = (object)array_merge((array)$this->attributes, (array)$this->parseAttributes($results['data']->upstream));
```

Comment:
And regarding the second bullet - you want that code to exist inside the `fetch` method, or `parseAttributes`?